packages/core/src/tests/createStore.test.ts

```diff
@@ -120,8 +121,31 @@ describe('createStore', () => {
             store.state$.subscribe();
             store.action$.next(someOtherAction);
 
+            await awaitNMockCalls(reducerMock, 4);
+
             expect(reducerMock).toBeCalledTimes(4);
-            expect(reducerMock).toHaveBeenLastCalledWith(initialState, someOtherAction);
+            expect(reducerMock).toHaveBeenLastCalledWith(initialState, someAction);
+        });
+
+        it('should send actions back to the action$ stream respecting dispatch order', async () => {
+            const aAction = { type: 'A' };
+            const bAction = { type: 'B' };
+            const someEffect: Effect<Action, {}> = action$ =>
+                action$.pipe(ofType(aAction.type), mapTo(bAction));
+
+            const reducerMock = jest.fn((state = initialState, _action) => state);
+
+            const store = createStore(reducerMock, someEffect);
+
+            store.state$.subscribe();
+            store.action$.next(aAction);
+
+            await awaitNMockCalls(reducerMock, 4);
```

Comment:
This is what I wanted to avoid, `asyncScheduler` makes everything async which might be not desirable if you are operating on sync effects. I was also experimenting with `queueScheduler` but I couldn't get it working properly.

Reply:
For the majority of cases, It probably won't matter though...

Reply:
Yup, had the same reflection... But it is definitely the simplest fix 🤷